document minimum required rust version in BUILD.md

Error information:

```bash
error[E0599]: no method named `eq_ignore_ascii_case` found for type `&str` in the current scope
  --> proxy/src/fully_qualified_authority.rs:56:19
   |
56 |             (part.eq_ignore_ascii_case("svc"), false)
   |                   ^^^^^^^^^^^^^^^^^^^^
   |
   = help: items from traits can only be used if the trait is in scope
   = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
           candidate #1: `use std::ascii::AsciiExt;`

error: aborting due to previous error

error: Could not compile `conduit-proxy`.
```

I run this build in ubuntu 16.04 with rustc 1.22.1. Should I upgrade to rustc 1.23 to build conduit?

--- a/BUILD.md
+++ b/BUILD.md
@@ -310,6 +310,8 @@ yarn karma start --single-run
 These commands assume a working [Rust](https://www.rust-lang.org)
 environment.
 
+Note that we _only_ support the most recent `stable` version of Rust.
+
 To build and run the Rust proxy:
 
 ```bash

--- a/proxy/Dockerfile
+++ b/proxy/Dockerfile
@@ -7,7 +7,7 @@
 # When PROXY_SKIP_TESTS is set and not empty, tests are not run. Otherwise, tests are run
 # against either unoptimized or optimized proxy code, according to PROXY_UNOPTIMIZED.
 
-ARG RUST_IMAGE=rust:1.23.0
+ARG RUST_IMAGE=rust:1.24.0
 ARG RUNTIME_IMAGE=gcr.io/runconduit/base:2017-10-30.01
 
 ## Builds the proxy as incrementally as possible.

